src/include/duckdb/storage/compression/alp/alp_constants.hpp

```diff
@@ -0,0 +1,134 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/storage/compression/alp/alp_constants.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/common/limits.hpp"
+#include "duckdb/main/config.hpp"
+
+namespace duckdb {
+
+class AlpConstants {
+public:
+	static constexpr uint32_t ALP_VECTOR_SIZE = 1024;
+	static constexpr uint32_t RG_SAMPLES = 8;
+	static constexpr uint16_t SAMPLES_PER_VECTOR = 32;
+	// We calculate how many equidistant vector we must jump within a rowgroup
+	static constexpr uint32_t RG_SAMPLES_DUCKDB_JUMP = (STANDARD_ROW_GROUPS_SIZE / RG_SAMPLES) / STANDARD_VECTOR_SIZE;
+
+	static constexpr uint8_t HEADER_SIZE = sizeof(uint32_t);
+	static constexpr uint8_t EXPONENT_SIZE = sizeof(uint8_t);
+	static constexpr uint8_t FACTOR_SIZE = sizeof(uint8_t);
+	static constexpr uint8_t EXCEPTIONS_COUNT_SIZE = sizeof(uint16_t);
+	static constexpr uint8_t EXCEPTION_POSITION_SIZE = sizeof(uint16_t);
+	static constexpr uint8_t FOR_SIZE = sizeof(uint64_t);
+	static constexpr uint8_t BIT_WIDTH_SIZE = sizeof(uint8_t);
+	static constexpr uint8_t METADATA_POINTER_SIZE = sizeof(uint32_t);
+
+	static constexpr uint8_t SAMPLING_EARLY_EXIT_THRESHOLD = 2;
+
+	static constexpr double COMPACT_BLOCK_THRESHOLD = 0.80;
+
+	// This is the highest int64 that can be exactly encoded as a double
```

Comment:
The same number that results from that query you quoted is the one we have as a constant. Which corresponds to (almost) `2^63`. Maybe I did not understand the question entirely.

Reply:
Right. The comment is indeed not accurate. It should be: `Largest double which fits into an int64`

In a nutshell, at some point we do a cast from `double` to `int64` and viceversa. This can return `undefined behaviour` for some special values: Infinites, NaN and when the double is higher than this value: `9223372036854774784`. Because once we pass this threshold the next possible `double` value is higher than the highest `int64`.